rpmlint/checks/PolkitCheck.py

```diff
@@ -0,0 +1,128 @@
+
+from rpmlint.checks.AbstractCheck import AbstractCheck
+
+
+class PolkitCheck(AbstractCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.polkit_privs_whitelist = config.configuration.get('PolkitPrivsWhitelist', ())
+        self.polkit_privs_files = config.configuration.get('PolkitPrivsFiles', ['/etc/polkit-default-privs.standard'])
+        self.privs = {}
+        self._collect_privs()
+
+    def _collect_privs(self):
+        for filename in self.polkit_privs_files:
+            if os.path.exists(filename):
+                self._parse_privs_file(filename)
+
+    def _parse_privs_file(self, filename):
+        with open(filename) as inputfile:
+            for line in inputfile:
+                line = line.split('#')[0].split('\n')[0]
+                if len(line):
+                    line = split(r'\s+', line)
+                    priv = line[0]
+                    value = line[1]
+                    self.privs[priv] = value
+
+    def check_perm_files(self, pkg):
+        """Checks files in polkit-default-privs.d."""
+        prefix = '/etc/polkit-default-privs.d/'
+        profiles = ('restrictive', 'standard', 'relaxed')
+        permfiles = []
+        # first pass, find additional files
+        for f in pkg.files:
+            if f.startswith(prefix):
+                if f in pkg.ghost_files:
+                    self.output.add_info('E', pkg, 'polkit-ghost-file', f)
+                    continue
+
+                bn = f[len(prefix):]
+                if bn not in self.polkit_privs_whitelist:
+                    self.output.add_info('E', pkg, 'polkit-unauthorized-file', f)
+
+                parts = bn.rsplit('.', 1)
+                if len(parts) == 2 and parts[-1] in profiles:
+                    bn = parts[0]
+
+                if bn not in permfiles:
+                    permfiles.append(bn)
+
+        for f in sorted(permfiles):
+            f = pkg.dirName() + prefix + f
+            for profile in profiles:
+                path = '.'.join((f, profile))
+                if os.path.exists(path):
+                    self._parse_privs_file(path)
+                    break
+            else:
+                self._parse_privs_file(f)
+
+    def check_actions(self, pkg):
+        """Checks files in the actions directory."""
+        prefix = '/usr/share/polkit-1/actions/'
+        for f in pkg.files:
+            # catch xml exceptions
+            try:
+                if f.startswith(prefix):
+                    if f in pkg.ghost_files:
+                        self.output.add_info('E', pkg, 'polkit-ghost-file', f)
+                        continue
+
+                    xml = parse(pkg.dirName() + f)
+                    for a in xml.getElementsByTagName('action'):
+                        self.check_action(pkg, a)
+            except Exception as x:
+                self.output.add_info('E', pkg, 'rpmlint-exception', f'{f:s} raised an exception: {x}')
```

Comment:
It's too much generic error. I would not catch an expection. It's caught in `lint.py`.

Reply:
I think it needs to catch the error to continue if the xml is broken. I would instead use `xml.parsers.expat.ExpatError`, would that be okay for you?

Reply:
Then, please rename the error to something like `polkit-xml-exception`?

Reply:
Here I meant reporting `self.output.add_info('E', pkg, 'polkit-xml-exception'...`

Reply:
Ah, sorry I misunderstood. It should be fixed now.